lib/routing/route.dart

```diff
@@ -48,14 +51,32 @@ class HeroDialogRoute<T> extends PageRoute<T> {
   String get barrierLabel => 'Popup dialog open';
 }
 
-void routeSubpage(Widget page, BuildContext context) {
-  Navigator.push(
-      context,
-      HeroDialogRoute(
-          builder: (context) =>
-              Hero(tag: "post-page", child: PointerInterceptor(child: page))));
-}
+/**
+ * Routes to a page that should be displayed overtop the current one
+ */
+void routeSubpage(Widget page, BuildContext context) => Navigator.push(
+    context,
+    HeroDialogRoute(
+        builder: (context) =>
+            Hero(tag: "post-page", child: PointerInterceptor(child: page))));
 
-void routePage(Widget page, BuildContext context) {
-  Navigator.push(context, MaterialPageRoute(builder: (context) => page));
-}
+/**
+ * Routes to another page
+ */
+void routePage(Widget page, BuildContext context) =>
+    Navigator.push(context, MaterialPageRoute(builder: (context) => page));
+
+/**
+ * Routes back to previous page from a subpage
+ */
+void routeBack(BuildContext context) => Navigator.pop(context);
+
+/**
+ * Routes back to the page in HomePage
+ * 0 -> MapPage
+ * 1 -> ListPage
+ * 2 -> ChatPage
+ * 3 -> ProfilePage
+ */
+void routeHome(int page, BuildContext context) => Navigator.push(context,
```

Comment:
I do like what you've done with the nav bar. It looks clean. 👍 
A slight nitpick we can handle later is the fact that pushing and popping gives an unsmooth feeling where if we click on nav bar we expect a smooth transition, but that may be transitions and not push and pop. Maybe make this an enhancement later on?

Reply:
Yep, I'll add this as an issue in the backlog

Reply:
#47